distage/distage-framework-docker/src/main/scala/izumi/distage/docker/FileLockMutex.scala

```diff
@@ -22,23 +22,18 @@ object FileLockMutex {
     F: DIEffect[F],
     P: DIEffectAsync[F],
   ): F[E] = {
-    def acquireAndRun(channel: FileChannel, attempts: Int): F[E] = {
-      F.maybeSuspend {
-          logger.debug(s"Attempt ${attempts -> "num"} out of $maxAttempts to acquire lock for $filename.")
-          try {
-            Option(channel.tryLock())
-          } catch {
-            case _: OverlappingFileLockException => None
-          }
-        }.flatMap {
-          case Some(lock) =>
-            effect.guarantee(F.maybeSuspend(lock.close()))
-          case None if attempts < maxAttempts =>
-            P.sleep(waitFor).flatMap(_ => acquireAndRun(channel, attempts + 1))
-          case None =>
+    def retryOnFileLock(eff: F[Option[FileLock]], attempts: Int = 0): F[Option[FileLock]] = {
```

Comment:
```suggestion
    def retryOnFileLock(eff: F[FileLock], attempts: Int = 0): F[Option[FileLock]] = {
```
might wrap into Some later since `eff` cannot return None now

Reply:
Done.